include/zephyr/drivers/can.h

```diff
@@ -1360,7 +1360,7 @@ static inline uint8_t can_dlc_to_bytes(uint8_t dlc)
 	static const uint8_t dlc_table[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 12,
 					    16, 20, 24, 32, 48, 64};
 
-	return dlc > 0x0F ? 64 : dlc_table[dlc];
+	return dlc_table[dlc > 0xF ? 16 : dlc];
```

Comment:
Wait. Don't we only have 16 elements, so that `dlc_table[16]` would actually be out of bounds? Why does it not complain now? A little bit too late today... Maybe magic number should better be replaced with `sizeof` calculation.

Reply:
dlc_table[MIN(dlc, sizeof(dlc_table) - 1)]

Reply:
I'm shocked.